business_objects/attribute.py

```diff
@@ -50,27 +50,51 @@ def get_all_by_names(project_id: str, attribute_names: List[str]) -> List[Attrib
     )
 
 
-def get_all(project_id: str) -> List[Attribute]:
-    return session.query(Attribute).filter(Attribute.project_id == project_id).all()
+def get_all(
+    project_id: str, state_filter: Optional[List[str]] = None
+) -> List[Attribute]:
+    if state_filter is None:
```

Comment:
caution, this is a bit misleading; as the user of this function, i would assume that setting `state_filter` to `None` will actually give me _all_ states.
another option would be to use the lists as the default argument (`state_filter: List[str] = [AttributeState.UPLOADED.value, AttributeState.USABLE.value]`).
generally, i would not recommend this, as the list is only created once as the function is defined (see https://docs.python-guide.org/writing/gotchas/), but in this case, it doesn't matter, as we never append to the default list.

just my two cents, handle however you like :)

Reply:
Changed to usage of the list as default argument